packages/wonder-blocks-layout/components/media-layout.test.js

```diff
@@ -0,0 +1,343 @@
+// @flow
+import React from "react";
+import {StyleSheet} from "aphrodite";
+import {View} from "@khanacademy/wonder-blocks-core";
+
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
+import MediaLayout from "./media-layout.js";
+import {resizeWindow, matchMedia} from "../util/test-util.js";
+
+const getArgsToChildren = async (styleSheets) =>
+    new Promise((resolve, reject) => {
+        mount(
+            <MediaLayout styleSheets={styleSheets}>
+                {({mediaSize, mediaSpec, styles}) => {
+                    resolve({mediaSize, mediaSpec, styles});
+                    return <View style={styles.test}>Hello, world!</View>;
+                }}
+            </MediaLayout>,
+        );
+    });
+
+const getStyle = (styleSheets) => {
+    const wrapper = mount(
+        <MediaLayout styleSheets={styleSheets}>
+            {({mediaSize, mediaSpec, styles}) => {
+                return <View style={styles.test}>Hello, world!</View>;
+            }}
+        </MediaLayout>,
+    );
+    const style = wrapper.find("div").prop("style");
+
+    return style;
+};
```

Comment:
Whereas in implementations, duplication is often a code smell, I find the opposite is true in testing since test cases are intended to be discrete, isolated scenarios. What do you think about inlining these things in each test instead of sharing implementation across tests? I think it would make each test case easier to grok in place as well as ease maintenance (especially if test cases diverge).